components/VercelChat/tools/browser/BrowserToolResult.tsx

```diff
@@ -0,0 +1,102 @@
+"use client";
+
+import { cn } from "@/lib/utils";
+import { ExtractResultView } from "./ExtractResultView";
+import { MessageResultView } from "./MessageResultView";
+import { ScreenshotView } from "./ScreenshotView";
+
+// Style constants
+const STYLES = {
+  text: { primary: "text-gray-900 dark:text-gray-100" },
+  bg: {
+    card: "bg-white dark:bg-gray-900",
+    error: "bg-gray-100 dark:bg-gray-800",
+  },
+  border: "border-gray-200 dark:border-gray-700",
+} as const;
+
+export interface BrowserToolResultType {
+  success: boolean;
+  error?: string;
+  sessionUrl?: string;
+  platformName?: string;
+  data?: unknown;
+  initialScreenshotUrl?: string;
+  finalScreenshotUrl?: string;
+  message?: string;
+  screenshotUrl?: string;
+}
+
+// Sub-component: Section header
+function SectionHeader({ title, className }: { title: string; className?: string }) {
+  return (
+    <div className={cn("pb-3", STYLES.border, "border-b", className)}>
+      <span className={cn("text-sm font-semibold", STYLES.text.primary)}>
+        {title}
+      </span>
+    </div>
+  );
+}
```

Comment:
Single Responsibility Principle
- actual: multiple functions defined in a single file
- required: exactly one function defined per file
- Note: you added a comment stating this is a `sub component`. The comment is a code smell highlighting this doesn't follow SRP.